packages/database/src/migrations/20210831002757-AddEntitiesRelationToSamoaForPenfaaSamoa-modifies-data.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+import { insertObject, codeToId, generateId, deleteObject } from '../utilities';
+import samoaHierarchyData from './migrationData/20210831002757-AddEntitiesRelationToSamoaForPenfaaSamoa/Entites_Hierarchy_Pen_Faa.json';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const projectCode = 'penfaa_samoa';
+
+const hierarchyNameToId = async (db, name) => {
+  const record = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = '${name}'`);
+  return record?.rows[0].id;
+};
+
+exports.up = async function (db) {
+  const entityHierarchyId = generateId();
+  await insertObject(db, 'entity_hierarchy', {
+    id: entityHierarchyId,
+    name: projectCode,
+    canonical_types: '{country,district,sub_district,school}',
```

Comment:
Not sure `sub_district` should be a canonical type here. They currently only exist in covid_samoa hierarchy (no parent_id) as defined in `entity_relation` table. So perhaps better to remove `sub_district` and define entity_relation rows for each sub_district with newly created `entityHierarchyId`.
I'm assuming we don't want the 'Unknown' entities in this hierarchy (something to check with @julianam-w ) so they would be not be added. If `Unknown Island` is not needed then `district` can remain canonical (else define similar to `covid_samoa` heirarchy.

Reply:
as a suggestion something like:
```
select * from entity_relation er 
join entity e on er.child_id = e.id and e."type" = 'sub_district'
where er.entity_hierarchy_id in (select eh.id from entity_hierarchy eh where eh."name" = 'covid_samoa');
```

Reply:
Please exclude all "Unknown" (district, sub_district, facility, village, household)

Reply:
So leaving `district` as canonical will exclude Unknown Island (as it has no parent_id on entity. The above query will exclude 'Unknown District' and facility, village, and household are not used in this hierarchy.